reference-adapter/pspcl-ifix-adapter/src/main/java/org/egov/repository/builder/PspclIfixQueryBuilder.java

```diff
@@ -0,0 +1,258 @@
+                param.addValue(PspclIfixQueryBuilder.BILNO, pspclPaymentDetail.getBILNO());
+                param.addValue(PspclIfixQueryBuilder.BILISSDT, pspclPaymentDetail.getBILISSDT());
+                param.addValue(PspclIfixQueryBuilder.DUEDTCASH, pspclPaymentDetail.getDUEDTCASH());
+                param.addValue(PspclIfixQueryBuilder.DUEDTCHQ, pspclPaymentDetail.getDUEDTCHQ());
+                param.addValue(PspclIfixQueryBuilder.STATUS_P, pspclPaymentDetail.getSTATUS_P());
+
+                param.addValue(PspclIfixQueryBuilder.CREATEDBY, pspclPaymentDetail.getAuditDetails().getCreatedBy());
+                param.addValue(PspclIfixQueryBuilder.CREATEDTIME, pspclPaymentDetail.getAuditDetails().getCreatedTime());
+                param.addValue(PspclIfixQueryBuilder.LASTMODIFIEDBY, pspclPaymentDetail.getAuditDetails().getLastModifiedBy());
+                param.addValue(PspclIfixQueryBuilder.LASTMODIFIEDTIME, pspclPaymentDetail.getAuditDetails().getLastModifiedTime());
+
+                String jsonData = "";
+                try {
+                    jsonData = objectMapper.writeValueAsString(pspclPaymentDetail.getPaymentJsonData());
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
+                }
+                param.addValue(PspclIfixQueryBuilder.JSON_DATA, jsonData, Types.OTHER);
+
+                sqlParameterSources.add(param);
+            }
+        }
+
+        return sqlParameterSources;
+    }
+
+    public List<SqlParameterSource> getEventPostingSqlParameterSources(List<EventPostingDetail> eventPostingDetailList) {
+        List<SqlParameterSource> sqlParameterSources = new ArrayList<>();
+        if (eventPostingDetailList != null && !eventPostingDetailList.isEmpty()) {
+            for (EventPostingDetail eventPostingDetail : eventPostingDetailList) {
+                MapSqlParameterSource param = new MapSqlParameterSource();
+                param.addValue(PspclIfixQueryBuilder.TENANTID, eventPostingDetail.getTenantId());
+                param.addValue(PspclIfixQueryBuilder.IFIXEVENTID, eventPostingDetail.getIfixEventId());
+                param.addValue(PspclIfixQueryBuilder.REFERENCEID, eventPostingDetail.getReferenceId());
+                param.addValue(PspclIfixQueryBuilder.EVENTTYPE, eventPostingDetail.getEventType());
+                param.addValue(PspclIfixQueryBuilder.STATUS, eventPostingDetail.getStatus());
+                param.addValue(PspclIfixQueryBuilder.ERROR, eventPostingDetail.getError());
+                param.addValue(PspclIfixQueryBuilder.LASTMODIFIEDDATE, eventPostingDetail.getLastModifiedDate());
+                param.addValue(PspclIfixQueryBuilder.CREATEDDATE, eventPostingDetail.getCreatedDate());
+
+                param.addValue(PspclIfixQueryBuilder.RECORD, eventPostingDetail.getRecord(), Types.OTHER);
+                sqlParameterSources.add(param);
+            }
+        }
+
+        return sqlParameterSources;
+    }
```

Comment:
Did this file become this long because we chose JdbcTemplate instead of JPA?

Reply:
yes